server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -698,53 +698,55 @@ private synchronized IndexService createIndexService(
         List<IndexEventListener> builtInListeners,
         IndexingOperationListener... indexingOperationListeners
     ) throws IOException {
-        final IndexSettings idxSettings = new IndexSettings(indexMetadata, settings, indexScopedSettings);
-        // we ignore private settings since they are not registered settings
-        indexScopedSettings.validate(indexMetadata.getSettings(), true, true, true);
-        logger.debug(
-            "creating Index [{}], shards [{}]/[{}] - reason [{}]",
-            indexMetadata.getIndex(),
-            idxSettings.getNumberOfShards(),
-            idxSettings.getNumberOfReplicas(),
-            indexCreationContext
-        );
+        try (var ignored = threadPool.getThreadContext().stashContext()) {
```

Comment:
I think we want to stay in the same context, not start a fresh one:

```suggestion
        try (var ignored = threadPool.getThreadContext().newStoredContext()) {
```